actionmailbox/Rakefile

```diff
@@ -12,6 +26,57 @@ Rake::TestTask.new do |t|
   t.verbose = true
 end
 
+file PRIVATE_KEY do |t|
+  require "openssl"
+  key = OpenSSL::PKey::RSA.new 2048
+  File.open(t.name, "w") do |f|
+    f.write key.to_pem
+  end
+end
+
+file CERTIFICATE => PRIVATE_KEY do |t|
+  require "openssl"
+  key = OpenSSL::PKey::RSA.new File.read(PRIVATE_KEY)
+  cert = OpenSSL::X509::Certificate.new
+  cert.version = 2
+  cert.serial = 2
+  cert.subject = OpenSSL::X509::Name.parse "/DC=org/DC=ruby-lang/CN=Ruby certificate"
+  cert.issuer = cert.subject # root CA is the issuer
+  cert.public_key = key.public_key
+  cert.not_before = Time.now
+  cert.not_after = cert.not_before + 1 * 365 * 24 * 60 * 60 # 10 years validity
+  ef = OpenSSL::X509::ExtensionFactory.new
+  ef.subject_certificate = cert
+  ef.issuer_certificate = cert
+  cert.add_extension(ef.create_extension("keyUsage", "digitalSignature", true))
+  cert.add_extension(ef.create_extension("subjectKeyIdentifier", "hash", false))
+  cert.sign(key, OpenSSL::Digest::SHA256.new)
+
+  File.open(t.name, "w") do |f|
+    f.write cert.to_pem
+  end
+end
+task certificates: [PRIVATE_KEY, CERTIFICATE]
+
+
+task sign_aws_fixtures: :certificates do
```

Comment:
When is this task called?

Reply:
Added description. Used for re-signing fixtures if they are modified.